obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+        E.
+    :type overwrite: bool
+    :param overwrite: force to overwrite old files, defaults to False
+
+    :returns: str: name of sfile written
+
+    .. note::
+
+        Seisan can find waveforms either by their relative or absolute \
+        path, or by looking for the file recursively in directories within \
+        the WAV directory in your seisan install.  Because all lines need to \
+        be less than 79 characters long (fortran hangover) in the s-files, \
+        you will need to determine whether the full-path is okay or not.
+    """
+    import datetime
+    import os
+    from obspy.core.event import Catalog, Event
+    # First we need to work out what to call the s-file and open it
+    # Check that user ID is the correct length
+    if len(userid) != 4:
+        raise IOError('%s User ID must be 4 characters long' % userid)
+    # Check that outdir exists
+    if not os.path.isdir(outdir):
+        raise IOError('Out path does not exist, I will not create this: ' +
+                      outdir)
+    # Check that evtype is one of L,R,D
+    if evtype not in ['L', 'R', 'D']:
+        raise IOError('Event type must be either L, R or D')
+    if explosion:
+        evtype += 'E'
+    # Check that there is one event
+    if isinstance(event, Catalog) and len(event) == 1:
+        event = event[0]
+    elif isinstance(event, Event):
+        event = event
+    else:
+        raise IOError('Needs a single event')
+    if not isinstance(wavefiles, list):
+        wavefiles = [str(wavefiles)]
+    # Determine name from origin time
+    try:
+        evtime = event.origins[0].time
+    except IndexError:
+        msg = 'list index out of range: Need at least one origin with at ' +\
+            'least an origin time'
+        raise IndexError(msg)
+    if not evtime:
+        msg = 'event has an origin, but time is not populated.  ' +\
+            'This is required!'
+        raise ValueError(msg)
+    # Attempt to cope with possible pre-existing files
+    if not filename:
+        range_list = []
+        for i in range(30):  # Look +/- 30 seconds around origin time
+            range_list.append(i)
+            range_list.append(-1 * i)
+        range_list = range_list[1:]
+        for add_secs in range_list:
+            sfilename = (evtime + add_secs).datetime.strftime('%d-%H%M-%S') +\
+                evtype[0] + '.S' + (evtime + add_secs).\
+                datetime.strftime('%Y%m')
+            if not os.path.isfile(outdir + os.sep + sfilename):
+                sfile = open(outdir + os.sep + sfilename, 'w')
+                break
+            elif overwrite:
+                sfile = open(outdir + os.sep + sfilename, 'w')
+                break
+        else:
+            raise IOError(outdir + os.sep + sfilename +
+                          ' already exists, will not overwrite')
+    else:
+        sfile = open(outdir + os.sep + filename, 'w')
+    # Write the header info.
+    if event.origins[0].latitude:
```

Comment:
@krischer You appear to have written exactly what's already there? I guess you mean `if event.origins[0].latitude is not None:`.